service/controller/v3/arm_templates/virtual_network_gateway.json

```diff
@@ -0,0 +1,103 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterID": {
+      "type": "string"
+    },
+    "GiantSwarmTags": {
+      "type": "object",
+      "defaultValue": {
+        "provider": "F80D01C0-7AAC-4440-98F6-5061511962AD"
+      }
+    },
+    "virtualNetworkName": {
+      "type": "string"
+    },
+    "vpnGatewayName": {
+      "type": "string"
+    },
+    "vpnSubnetName": {
+      "type": "string",
+      "defaultValue": "GatewaySubnet"
+    },
+    "virtualNetworksGatewayAPIVersion": {
+      "type": "string",
+      "defaultValue": "2017-10-01",
+      "metadata": {
+        "description": "API version used by the Microsoft.Network/virtualNetworkGateways resource."
+      }
+    },
+    "publicIPAddressesAPIVersion": {
+      "type": "string",
+      "defaultValue": "2017-10-01",
+      "metadata": {
+        "description": "API version used by the Microsoft.Network/publicIPAddresses resource."
+      }
+    }
```

Comment:
`*APIVersion` should be variables instaed of parameters. I'd say the same for the `vpnSubnetName`.

Reply:
done